index.html

```diff
@@ -396,49 +392,37 @@ <h3>
           </h3>
           <p data-tests=
           "getCurrentPosition_IDL.https.html, getCurrentPosition_TypeError.html">
-            The <dfn>getCurrentPosition()</dfn> method takes one, two or three
-            arguments. When called, it MUST immediately return and then
-            asynchronously attempt to obtain the current location of the
-            device. If the attempt is successful, the
-            |successCallback:PositionCallback| MUST be invoked with a new
-            {{GeolocationPosition}} object, reflecting the current location of
-            the device. If the attempt fails, the
-            |errorCallback:PositionErrorCallback| MUST be invoked with a new
-            {{GeolocationPositionError}} object, reflecting the reason for the
-            failure.
-          </p>
-          <p>
-            The implementation of the {{Geolocation/getCurrentPosition()}}
-            method MUST execute the following set of steps:
+            When {{Geolocation/getCurrentPosition()}} is invoked, the user
+            agent MUST:
           </p>
           <ol>
-            <li>Return and continue asynchronously.
+            <li>Return and continue [=in parallel=].
             </li>
             <li>If the <a>environment settings object</a> is a <a>non-secure
-            context</a> or the <a>current settings object</a>'s [=environment
-            settings object / responsible document=] is not [=allowed to use=]
-            the "geolocation" feature:
-              <ol>
-                <li>If |errorCallback| is not null, invoke the |errorCallback|
-                with a newly created {{GeolocationPositionError}} object whose
-                {{GeolocationPositionError/code}} attribute is initialized to
-                {{GeolocationPositionError/PERMISSION_DENIED}}.
-                </li>
-                <li>Terminate this algorithm.
-                </li>
-              </ol>
+            context</a>, then [=call back with error=] |errorCallback| and
+            {{GeolocationPositionError/PERMISSION_DENIED}} and terminate this
+            algorithm.
+            </li>
+            <li>[=Request permission to use=] "geolocation" (see
```

Comment:
You need to capture the result of this call, which will be "granted" or "denied", and fail if it returns "denied".

Reply:
Yeah, I was trying to be clever and just let "allowed to use" deal with it. I'll do as you suggest.